cpp/CMakeLists.txt

```
@@ -191,8 +191,11 @@ endif(BUILD_TESTS)
 ###################################################################################################
 # - include paths ---------------------------------------------------------------------------------
 
-include_directories("${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}"
-                    "${CMAKE_BINARY_DIR}/include"
+if(NOT("${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}" STREQUAL ""))
```

I think it would be more idiomatic and clearer to just say 
```suggestion
if(CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES)
```

https://cmake.org/cmake/help/v3.0/command/if.html